### DISCUSSION TITLE
Special characters e.g. é in event text causes the ics file to not be generated correctly

### DISCUSSION THREAD

**Discussion by mmodha** [markuspoerschke/iCal#258] [Q&A] [answered]
The ics file becomes invalid because the file will be shorted e.g. to 'END:VCALEND', this is because of special characters in the summary text. For each special character a letter will be removed.

**Comment by mmodha** [comment]
Hi any ideas when this issue can this be looked at. Thanks

**Comment by orditeck** [comment, marked as answer]
I don't experience this issue, I'm using special characters in calendar title, eg.:
```php
new \Eluceo\iCal\Component\Calendar("Exemple avec des èàù accents")
```
and events, eg.:
```php
(new \Eluceo\iCal\Component\Event())->setSummary("Résidus verts")
```

If I were you I'd look elsewhere, eg., setting the right headers:
```php
header('Content-Type: text/calendar; charset=utf-8');
```

**Comment by markuspoerschke** [comment]
@mmodha Can you check if the advice by @orditeck works?

**Comment by bartvanhoutte** [comment]
This might be the issue: https://support.google.com/calendar/thread/204991361/google-calendar-not-importing-ics-files-correctly-requires-unnecessary-bom?hl=en.

Adding a BOM fixed it for me: 

```php
  private function streamCalendar(Calendar $calendar): void {
    $handle = fopen('php://output', 'r+');

    $BOM = chr(0xEF) . chr(0xBB) . chr(0xBF);
    fwrite($handle, $BOM);

    $calendarComponent = (new CalendarFactory)->createCalendar($calendar);

    foreach ($calendarComponent as $line) {
      fputs($handle, (string) $line);
    }

    fclose($handle);
  }
```
